rfcs/require-by-string.md

```diff
@@ -0,0 +1,360 @@
+
+```lua
+-- Beginning of /Users/JohnDoe/LuauModules/Math/math.luau
+local sqrt = require("../MathHelperFunctions/sqrt")
+```
+
+If we then launched the Luau CLI from the directory `/Users/JohnDoe/Projects/MyCalculator` and required `math.luau` as follows:
+
+```lua
+> local math = require("/Users/JohnDoe/LuauModules/Math/math")
+```
+
+This would cause the following:
+- The current implementation of require would successfully find `math.luau`, as its absolute path was given.
+- Then, it would execute the contents of `math.luau` from the context of the current working directory `/Users/JohnDoe/Projects/MyCalculator`.
+- When attempting to require `sqrt.luau` from `math.luau`, instead of looking in the directory `/Users/JohnDoe/LuauModules/MathHelperFunctions`, the relative path will be evaluated in relation to the current working directory.
+- This will look for `sqrt.luau` in `/Users/JohnDoe/Projects/MathHelperFunctions`, which is a directory that may not even exist.
+
+This behavior is [problematic](https://github.com/Roblox/luau/issues/959), and puts an unnecessary emphasis on the directory from which the Luau CLI is running. A better solution is to evaluate relative paths in relation to the file that is requiring them.
+
+
+### Package management
+
+While package management itself is outside of the scope of this RFC, we want to make it possible for a package management solution to be developed in a way that integrates well with our require syntax.
+
+To require a Luau module under the current implementation, we must require it either by relative or absolute path:
+
+```lua
+-- Requiring Roact by absolute path
+local Roact = require("C:/LuauModules/Roact-v1.4.2")
+```
+
+If we wanted to change the location of this package on disk, we would have to modify all files that require the old path above, which essentially boils down to manual package management. This approach is tedious and prone to mistakes.
+
+To solve this, we introduce aliases in this RFC, which would allow for future package management software to maintain package paths and versions by directly modifying alias maps.
+
+This would also create simple and readable require statements for developers.
+
+```lua
+-- Suppose "@src" is an alias for the same directory as "../../../../../"
+
+-- Instead of this:
+local result = require("../../../../../HelperModules/graphing")
+
+-- We could have this:
+local result = require("@src/HelperModules/graphing")
+```
+
+## Design
+
+### String syntax
+
+Our require syntax should:
+
+- Allow a library written in Luau to be imported into the Roblox engine and "just work".
+- Support compile-time file resolution _where possible_ for type checking.
+- Be consistent across platforms and both inside and outside of Roblox.
+- Provide support for binding unambiguous aliases to relative and absolute paths.
+- Attempt to remain consistent with the existing Luau path syntax.
+
+For compatibility across platforms, we will automatically map `/` onto `\`.
```

Comment:
This is one approach for compatibility with Windows: any path given to `require` would have its backslashes replaced with forward slashes before resolving the path to a file (since Windows FS APIs support both separators). By consolidating these into one, we also ensure that `C:/absolute\path/to\my/file` and `C:\absolute\path\to\my\file` are stored under the same name in the module cache.

However, this would also mean that the output of `debug.info` would have forward slashes on Windows. I suppose we could separate the chunkname given to `luau_load` from the name used for caching.